module/VuFind/src/VuFind/Config/Upgrade.php

```diff
@@ -709,6 +709,13 @@ protected function upgradeConfig()
             unset($newConfig['Syndetics']['url']);
         }
 
+        // Convert spellchecker 'simple' option
+        if (isset($newConfig['Spelling']['simple']) && !isset($newConfig['Spelling']['dictionaries'])) {
```

Comment:
The unit test I added fails because this `if` never resolves to true.  Looks like an earlier part of the upgrade process starts with the latest config.ini, so there *is* a (default) `dictionaries` element there already, and the second condition fails.  I added that check to avoid overriding any actual configured dictionaries present in the local config, which I still think is a concern, but not sure now.

Reply:
Implemented; this made sense.